cpp/include/cudf/table/table_view.hpp

```diff
@@ -141,6 +141,19 @@ class table_view_base {
  *---------------------------------------------------------------------------**/
 class table_view : public detail::table_view_base<column_view> {
   using detail::table_view_base<column_view>::table_view_base;
+
+ public:
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a table_view with set of specified columns.
+   *
+   * @throws cudf::logic_error
+   * If any element in `column_indices` is outside [0, num_columns())
+   *
+   * @param column_indices Indices of columns in the table
+   * @return A table_view consisting of columns from the original table
+   * specified by the elements of `column_indices`
+   *---------------------------------------------------------------------------**/
+  table_view select(std::vector<cudf::size_type> const& column_indices) const;
```

Comment:
@trevorsm7 can you break this into a separate PR so it can be used in https://github.com/rapidsai/cudf/pull/3250?